File: TMessagesProj/src/main/java/ua/itaysonlab/catogram/CatogramPreferencesNavigator.java
package ua.itaysonlab.catogram;

import org.telegram.ui.ActionBar.BaseFragment;

import ua.itaysonlab.catogram.preferences.AppearancePreferencesEntry;
import ua.itaysonlab.catogram.preferences.ChatsPreferencesEntry;
import ua.itaysonlab.catogram.preferences.MainPreferencesEntry;
import ua.itaysonlab.catogram.preferences.SecurityPreferencesEntry;
import ua.itaysonlab.tgkit.TGKitSettingsFragment;

public class CatogramPreferencesNavigator {
    public static TGKitSettingsFragment createMainMenu() {
        return new TGKitSettingsFragment(new MainPreferencesEntry());
    }

    public static BaseFragment createChats() {
        return new TGKitSettingsFragment(new ChatsPreferencesEntry());
    }

    public static BaseFragment createAppearance() {
        return new TGKitSettingsFragment(new AppearancePreferencesEntry());
    }

    public static BaseFragment createSecurity() {
        return new TGKitSettingsFragment(new SecurityPreferencesEntry());
    }
}

File: TMessagesProj/src/main/java/ua/itaysonlab/catogram/CatogramPreferencesNavigator.kt
package ua.itaysonlab.catogram

import ua.itaysonlab.catogram.preferences.*
import ua.itaysonlab.tgkit.TGKitSettingsFragment

object CatogramPreferencesNavigator {
    @JvmStatic
    fun createMainMenu() = TGKitSettingsFragment(MainPreferencesEntry())

    fun createChats() = TGKitSettingsFragment(ChatsPreferencesEntry())
    fun createAppearance() = TGKitSettingsFragment(AppearancePreferencesEntry())
    fun createSecurity() = TGKitSettingsFragment(SecurityPreferencesEntry())

    fun createDB() = TGKitSettingsFragment(DoubleBottomPreferencesEntry())
}